Tasks/DotNetCoreCLI/restorecommand.ts

```diff
@@ -0,0 +1,168 @@
+import * as tl from "vsts-task-lib/task";
+import * as Q  from "q";
+import * as utility from './Common/utility';
+import locationHelpers = require("nuget-task-common/LocationHelpers");
+import * as auth from "nuget-task-common/Authentication";
+import {NuGetConfigHelper2} from "nuget-task-common/NuGetConfigHelper2";
+import peParser = require('nuget-task-common/pe-parser/index');
+import * as path from "path";
+import {VersionInfo} from "nuget-task-common/pe-parser/VersionResource";
+import { IPackageSource } from "nuget-task-common/Authentication";
+import {IExecOptions} from "vsts-task-lib/toolrunner";
+import * as nutil from "nuget-task-common/Utility";
+import * as commandHelper from "nuget-task-common/CommandHelper";
+
+
+const NUGET_ORG_V3_URL: string = "https://api.nuget.org/v3/index.json";
+
+export async function run(): Promise<void> {
+    let buildIdentityDisplayName: string = null;
+    let buildIdentityAccount: string = null;
+
+    try {
+        const projectSearch = tl.getDelimitedInput("projects", "\n", false);
+
+        // if no projectSearch strings are given, use "" to operate on the current directory
+        const projectFiles = projectSearch ? utility.getProjectFiles(projectSearch) : [""];
+
+        const noCache = tl.getBoolInput("noCache");
+        const verbosity = tl.getInput("verbosityRestore");
+        let packagesDirectory = tl.getPathInput("packagesDirectory");
+        if (!tl.filePathSupplied("packagesDirectory")) {
+            packagesDirectory = null;
+        }
+
+        // Setting up auth-related variables
+        tl.debug('Setting up auth');
+        const serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
+        let urlPrefixes = await locationHelpers.assumeNuGetUriPrefixes(serviceUri);
+        tl.debug(`Discovered URL prefixes: ${urlPrefixes}`);
+
+        // Note to readers: This variable will be going away once we have a fix for the location service for
+        // customers behind proxies
+        let testPrefixes = tl.getVariable("DotNetCoreCLITask.ExtraUrlPrefixesForTesting");
+        if (testPrefixes) {
+            urlPrefixes = urlPrefixes.concat(testPrefixes.split(";"));
+            tl.debug(`All URL prefixes: ${urlPrefixes}`);
+        }
+
+        let accessToken = auth.getSystemAccessToken();
```

Comment:
Should all of the config code placed in a common location since it's the same as nugetRestore?